pages/changelog/2024-11-20-multi-modal-custom-attachments.mdx

```diff
@@ -0,0 +1,69 @@
+---
+date: 2024-11-20
+title: Full multi-modal support, including audio, images, and arbitrary files
+badge: Launch Week 2 🚀
+description: Add arbitrary attachments to LLM traces in Langfuse to observe and analyze multi-modal LLM applications.
+author: Hassieb
+---
+
+import { ChangelogHeader } from "@/components/changelog/ChangelogHeader";
+
+<ChangelogHeader />
+
+[Back in August](/changelog/2024-08-21-openai-multi-modal-traces), we added support for rendering multi-modal traces in Langfuse if they reference external files. Since then, extending multi-modal support to base64 encoded images and arbitrary attachments has been one a top requests from the community ([GitHub Discussion thread](https://github.com/orgs/langfuse/discussions/3004) with 45 upvotes).
+
+Today, on **Day 3 of [Langfuse Launch Week 2](/blog/2024-11-17-launch-week-2)**, we are excited to share that full multi-modal support is now available in Langfuse Tracing. Read on to learn more about the scope and how it works.
+
+## What is supported?
+
+- Files: Images, audio, video, and any other arbitrary files
+- UI
+  - Audio is playable in-line; if transcripts are included, they are shown right next to the audio player
+  - Images are rendered in-line
+  - Other media types open in a new tab
+- Use in traces
+  - In-line in trace/observation inputs, outputs, and metadata
+  - As attachments to the trace or observation
+- Limits
+  - ???
```

Comment:
Not yet supporting LangfuseMedia attachments directly in decorated function args / kwargs or return value